undertow/src/main/java/org/wildfly/extension/undertow/HostAdd.java

```diff
@@ -117,6 +123,28 @@ protected void performRuntime(OperationContext context, ModelNode operation, Mod
                 sb.install();
             }
         }
+
+        if (activeReqTrackingEnabled) {
+            if (statisticsEnabled) {
+                Predicate predicate = null;
+                ModelNode predicateNode = HostDefinition.ACTIVE_REQUEST_TRACKING_PREDICATE.resolveModelAttribute(context, model);
+                if(predicateNode.isDefined()) {
+                    predicate = Predicates.parse(predicateNode.asString(), getClass().getClassLoader());
+                }
+                final CapabilityServiceBuilder<?> sb = context.getCapabilityServiceTarget()
+                        .addCapability(HostDefinition.ACTIVE_REQUEST_TRACKING_CAPABILITY);
+                final Consumer<ActiveRequestTrackerService> serviceConsumer =
+                        sb.provides(HostDefinition.ACTIVE_REQUEST_TRACKING_CAPABILITY,
+                        UndertowService.activeRequestTrackingServiceName(serverName, name));
```

Comment:
Why provide the same service using 2 service names?
It looks this this was done to workaround the incorrect dynamic name mapper defined by ACTIVE_REQUEST_TRACKING_CAPABILITY.

Reply:
We have a reference to the Host instance in this method already (line 101).
Therefore, we don't need a service to resolve the host in order to call Host.setActiveRequestTrackerHandler(...).

Reply:
Good point.